src/ring_theory/maps.lean

```diff
@@ -25,174 +26,64 @@ All types defined in this file are given a coercion to the underlying function.
 Ring isomorphism, automorphism, antihomomorphism, antiisomorphism, antiautomorphism, involution
 -/
 
-variables {R : Type*} {F : Type*}
-
-/- The Proposition that a function from a ring to a ring is an antihomomorphism -/
-class is_ring_anti_hom [ring R] [ring F] (f : R → F) : Prop :=
-(map_one [] : f 1 = 1)
-(map_mul [] : ∀ {x y : R}, f (x * y) = f y * f x)
-(map_add [] : ∀ {x y : R}, f (x + y) = f x + f y)
-
-namespace is_ring_anti_hom
-
-variables [ring R] [ring F] (f : R → F) [is_ring_anti_hom f]
-
-@[priority 100] -- see Note [lower instance priority]
-instance : is_add_group_hom f :=
-{ to_is_add_hom := ⟨λ x y, is_ring_anti_hom.map_add f⟩ }
-
-lemma map_zero : f 0 = 0 :=
-is_add_group_hom.map_zero f
-
-lemma map_neg {x} : f (-x) = -f x :=
-is_add_group_hom.map_neg f x
-
-lemma map_sub {x y} : f (x - y) = f x - f y :=
-is_add_group_hom.map_sub f x y
-
-end is_ring_anti_hom
-
-variables (R F)
+variables (R : Type*) (F : Type*)
 
 namespace ring_equiv
 
 open ring_equiv
 
-variables {R F} [ring R] [ring F] (Hs : R ≃+* F) (x y : R)
+variables {R F} [semiring R] [semiring F] (Hs : R ≃+* F)
 
 lemma bijective : function.bijective Hs :=
 Hs.to_equiv.bijective
 
-lemma map_zero_iff {x : R} : Hs x = 0 ↔ x = 0 :=
+lemma map_zero_iff (x : R) : Hs x = 0 ↔ x = 0 :=
```

Comment:
Why do you wan an explicit `x` here?

Reply:
Sorry, committing that change was a mistake, made while trying to diagnose a `simp` failure.